lib/request_handler/body_parser.rb

```diff
@@ -25,7 +25,7 @@ def run
 
     def flattened_request_body
       body = request_body.fetch('data') do
-        raise ExternalArgumentError, body: 'must contain data'
+        raise BodyParamsError, body: 'must contain data'
```

Comment:
Also in a later PR we can deal with reducing the redundancy in the error messages we introduce here